pkg/model/bootstrapscript.go

```diff
@@ -58,6 +58,21 @@ func (b *BootstrapScript) KubeEnv(ig *kops.InstanceGroup) (string, error) {
 	return string(data), nil
 }
 
+// KubeKeys returns the nodeup config for the instance group
+func (b *BootstrapScript) KubeKeys(ig *kops.InstanceGroup, cluster *kops.Cluster) (string, error) {
+	config, err := b.NodeUpConfigBuilder(ig)
+	if err != nil {
+		return "", err
+	}
+
+	if kops.CloudProviderID(cluster.Spec.CloudProvider) == kops.CloudProviderAWS && config.ConfigBase != nil {
+		configBaseValue := *(config.ConfigBase)
+		return fmt.Sprintf(`/usr/local/bin/aws s3 cp %s/pki/ssh/public/admin/ /tmp/admin --recursive --include "*"`, configBaseValue), nil
```

Comment:
That's very specific to the current aws cli - and I'm not sure how this behaves on custom AMI's. Not sure if another logic would make more sense, like download it using the aws go sdk in nodeup instead of using aws cli as a third party tool.

Reply:
will look at the api way for pulling the keys file